pkg/reconciler/inmemorychannel/controller/controller.go

```diff
@@ -140,10 +149,88 @@ func NewController(
 		Handler:    controller.HandleAll(globalResync),
 	})
 
+	// Enqueue the InMemoryChannel, if we have an EventPolicy which was referencing
+	// or got updated and now is referencing the InMemoryChannel
+	eventPolicyInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			enqueueApplyingChannelOfEventPolicy(inmemorychannelInformer.Lister(), obj, impl.EnqueueKey)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			// Here we need to check if either the old or the new EventPolicy was referencing the InMemoryChannel
+
+			alreadyEnqueued := enqueueApplyingChannelOfEventPolicy(inmemorychannelInformer.Lister(), oldObj, impl.EnqueueKey)
+			if !alreadyEnqueued {
+				enqueueApplyingChannelOfEventPolicy(inmemorychannelInformer.Lister(), newObj, impl.EnqueueKey)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			enqueueApplyingChannelOfEventPolicy(inmemorychannelInformer.Lister(), obj, impl.EnqueueKey)
+		},
+	})
```

Comment:
Wondering, if we can simplify this somehow (e.g. with a tracker instead) :thinking: 

I did it that way so far, because:
* It enqueues only the referenced IMC(s) from the event policy (not a "global-resync)
* It also enqueues the old referenced IMC, if the event policy got updated and the new version does not reference the IMC anymore